### PR TITLE
Sample Scene Bot, CharacterConfig Updates

### PR DESCRIPTION
**Pull request by vontell**
Updates the existing bots to use the new characterConfig type (rather than characterType). Also adds a new SampleScene bot that is used in the tutorial.

### CODE REVIEW

**Review by RG-nAmKcAz — APPROVED** [review]

File: abilitybot/startScenarioExample.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+    }; // healer
 }
 
 export async function startScenario(rg) {
 
-    const charType = CharInfo.type.indexOf(rg.characterType);
+    let charType = CharInfo.type.indexOf(rg.characterConfig.characterType);
+    const overrideType = JSON.parse(rg.characterConfig).characterType;
+    if (overrideType) {
+        charType = CharInfo.type.indexOf(overrideType);
```
Comment (abeizer) [inline abilitybot/startScenarioExample.js:14]: Shouldn't all of this just be
```
const characterType = CharInfo.type.indexOf(rg.characterConfig.characterType);
```

you don't need to parse the JSON because it should already be an object on `rg`

File: abilitybot/index.js [hunk 2]
```diff
@@ -5,13 +5,19 @@ let lastEnemyId = -1;
...
 }
 
 export async function processTick(rg) {
 
   // The character type we request may not be the one we actually get
-  charType = CharInfo.type.indexOf(rg.characterType);
+  const characterType = rg.characterConfig.characterType;
+  if (characterType) {
+    charType = CharInfo.type.indexOf(characterType);
+    console.log(`Unity bot configureBot function called, charType: ${charType} - characterType: ${characterType}`);
```
Comment (abeizer) [inline abilitybot/index.js:19]: This log isn't accurate

File: abilitybot/startScenarioExample.js [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+        console.log(`Unity bot configureBot function called, charType: ${charType} - characterType: ${characterType}`);
```
Comment (abeizer) [inline abilitybot/startScenarioExample.js:15]: This log isn't accurate

File: rgsamplebot/index.js
Comment (abeizer) [inline rgsamplebot/index.js:1]: Technically these are all sample bots so I'm not a fan of this name. Also this one wouldn't work in BossRoom - why is this here? Should this be part of the documentation instead?

**Review by abeizer — CHANGES_REQUESTED** [review]

File: abilitybot/startScenarioExample.js (hunk 1 above)
Comment (vontell) [reply abilitybot/startScenarioExample.js:14]: Good catch! Removed since this was just a duplicate anyways!

File: abilitybot/index.js (hunk 2 above)
Comment (vontell) [reply abilitybot/index.js:19]: Updated!

File: abilitybot/startScenarioExample.js (hunk 3 above)
Comment (vontell) [reply abilitybot/startScenarioExample.js:15]: Removed

File: rgsamplebot/index.js
Comment (vontell) [reply rgsamplebot/index.js:1]: Removed! Will have that be part of another project

**Comment by vontell** [comment]
@abeizer back to you!

File: abilitybot/index.js [hunk 4]
```diff
@@ -5,13 +5,19 @@ let lastEnemyId = -1;
...
 }
 
 export async function processTick(rg) {
 
   // The character type we request may not be the one we actually get
-  charType = CharInfo.type.indexOf(rg.characterType);
+  const characterType = rg.characterConfig.characterType;
+  if (characterType) {
+    charType = CharInfo.type.indexOf(characterType);
+    console.log(`Character type has been override, updating - characterType: ${characterType}`);
```
Comment (abeizer) [inline abilitybot/index.js:19]: This isn't accurate. Since you're defining the characterConfig for this bot in the player code, it will always be defined in the `rg` object - this doesn't necessarily mean that it was overridden during handshake

Comment (vontell) [reply abilitybot/index.js]: Fixed!

**Review by abeizer — APPROVED** [review]
